### PR TITLE
Python env upgrade - 2024

### PR DESCRIPTION
**Pull request by addyess**
Creates a python3.8 compatible requirements.txt for the CI

Key Features:
* removes ansible python requirement (it wasn't being used by any jobs -- the infra job pip installs ansible itself)
   - this  shrinks the pip environment by half (700MB -> 320MB)
* upgrades pyyaml away from the insanity where it needed pinned cython sources.
* Uses newer black

### CODE REVIEW

**Comment by addyess** [comment]
thanks @Cynerva

File: requirements.in [hunk 1]
```diff
@@ -1,5 +1,3 @@
-ansible<7.0.0; python_version<"3.9"
-ansible>=7.0.0; python_version>="3.9"
```
Comment (kwmonroe) [inline requirements.in:2]: :tada:

File: jobs/infra/fixtures/cleanup-env.sh [hunk 2]
```diff
@@ -35,7 +35,7 @@ docker image prune -a --filter until=24h --force
 docker container prune --filter until=24h --force
 rm -rf /var/lib/jenkins/venvs
 rm -rf /var/lib/jenkins/.tox
-tmpreaper 5h /tmp
+sudo tmpreaper 5h /tmp
```
Comment (kwmonroe) [inline jobs/infra/fixtures/cleanup-env.sh:37]: "just black" you say?  must be that new AI black catching formatting errors like this :)

File: jobs/infra/fixtures/cleanup-env.sh [hunk 3]
```diff
@@ -30,7 +30,6 @@ juju controllers --format json | jq -r '.controllers | keys[]' | parallel --ungr
 # done
 
 sudo apt clean
-sudo rm -rf /var/log/*
```
Comment (kwmonroe) [inline jobs/infra/fixtures/cleanup-env.sh:33]: wow.  why we had this in the first place is... troubling.

**Review by kwmonroe — APPROVED** [review]
LGTM!

**Pull request merged**